wolfcrypt/src/aes.c

```diff
@@ -3675,6 +3759,14 @@ int wc_AesSetIV(Aes* aes, const byte* iv)
         if (sz == 0) {
```

Comment:
Tempting.  But that would cause a change in behavior that might break something for someone.  There's a bunch of gated acceleration paths that are passing on `sz`, not `blocks`, and might have their own ideas of whether and how to error on short unaligned input.  (The `WOLFSSL_AES_CBC_ALIGNMENT_CHECKS` path will catch it for those that want/need it consistently caught, and you've well and truly convinced me that gating the check is the only safe way forward.)

We could chase down the logic of all those alternate paths, but why bother?  Only callers breaking the rules and passing in unaligned lengths would be affected by the change, which is probably a good reason not to bother changing it in the first place.